src/main/java/com/arpnetworking/metrics/common/sources/FileSource.java

```diff
@@ -73,6 +80,14 @@ public void stop() {
                     .setThrowable(e)
                     .log();
         }
+        try {
+            _lineProcessorExecutor.awaitTermination(10, TimeUnit.SECONDS);
```

Comment:
Oh you do await. My concern here is to ensure all data read from the file is processed so that we can do proper check pointing on shutdown. If we drop work at various stages it's going to make this MUCH harder.